examples/animation/animation_masks.rs

```diff
@@ -0,0 +1,365 @@
+            let prefix: Vec<_> = mask_group_prefix.split('/').map(Name::new).collect();
+            let suffix: Vec<_> = mask_group_suffix.split('/').map(Name::new).collect();
+
+            // Add each bone in the chain to the appropriate mask group.
+            for chain_length in 0..=suffix.len() {
+                let animation_target_id = AnimationTargetId::from_names(
+                    prefix.iter().chain(suffix[0..chain_length].iter()),
+                );
+                animation_graph
+                    .add_target_to_mask_group(animation_target_id, mask_group_index as u32);
+            }
+        }
+
+        // We're doing constructing the animation graph. Add it as an asset.
+        let animation_graph = animation_graphs.add(animation_graph);
+        commands.entity(entity).insert(animation_graph);
+
+        // Finally, play the animation.
+        player.play(node_index).repeat();
+    }
+}
+
+// A system that handles requests from the user to toggle mask groups on and
+// off.
+fn handle_button_toggles(
+    mut interactions: Query<
+        (
+            &Interaction,
+            &mut MaskGroupControl,
+            &mut BackgroundColor,
+            &Children,
+        ),
+        Changed<Interaction>,
+    >,
+    mut texts: Query<&mut Text>,
+    mut animation_players: Query<(&Handle<AnimationGraph>, &AnimationPlayer)>,
+    mut animation_graphs: ResMut<Assets<AnimationGraph>>,
+) {
+    for (interaction, mut mask_group_control, mut button_background_color, children) in
+        interactions.iter_mut()
+    {
+        // We only care about press events.
+        if *interaction != Interaction::Pressed {
+            continue;
+        }
+
+        // Toggle the state of the mask.
+        mask_group_control.enabled = !mask_group_control.enabled;
+
+        // Update the background color of the button.
+        button_background_color.0 = if mask_group_control.enabled {
+            Color::WHITE
+        } else {
+            Color::BLACK
+        };
+
+        // Update the text color of the button.
+        for &kid in children.iter() {
+            if let Ok(mut text) = texts.get_mut(kid) {
+                for section in &mut text.sections {
+                    section.style.color = if mask_group_control.enabled {
+                        Color::BLACK
+                    } else {
+                        Color::WHITE
+                    };
+                }
+            }
+        }
+
+        // Now grab the animation player. (There's only one in our case, but we
+        // iterate just for clarity's sake.)
+        for (animation_graph_handle, animation_player) in animation_players.iter_mut() {
+            // The animation graph needs to have loaded.
+            let Some(animation_graph) = animation_graphs.get_mut(animation_graph_handle) else {
+                continue;
+            };
+
+            // Grab the animation graph node that's currently playing.
```

Comment:
I'm new to animation and am a bit confused by this part.
Basically there is only one node with a playing animation, which is the root node of the `AnimationGraph`?